addons/arsenal/functions/fnc_buttonCargo.sqf

```diff
@@ -16,77 +16,92 @@
-        _maxLoad = gettext (configfile >> "CfgWeapons" >> vest GVAR(center) >> "ItemInfo" >> "containerClass");
-        _items = vestItems GVAR(center);
-        GVAR(currentItems) set [16,_items];
+        uniformContainer GVAR(center)
     };
+    // Vest
+    case IDC_buttonVest: {
+        if (_addOrRemove) then {
+            for "_i" from 1 to ([1, 5] select GVAR(shiftState)) do {
+                GVAR(center) addItemToVest _item;
+            };
+        } else {
+            for "_i" from 1 to ([1, 5] select GVAR(shiftState)) do {
+                GVAR(center) removeItemFromVest _item;
+            };
+        };
 
-    case IDC_buttonBackpack : {
-        if (_addOrRemove > 0) then {
-                for "_count" from 1 to ([1, 5] select (GVAR(shiftState))) do {
-                    GVAR(center) addItemToBackpack _item;
-                };
-            } else {
-                for "_count" from 1 to ([1, 5] select (GVAR(shiftState))) do {
-                    GVAR(center) removeItemFromBackpack _item;
-                };
+        _containerItems = vestItems GVAR(center);
+
+        // Update currentItems
+        GVAR(currentItems) set [IDX_CURR_VEST_ITEMS, ((getUnitLoadout GVAR(center)) select IDX_LOADOUT_VEST) param [1, []]];
+
+        // Update load bar
+        (_display displayCtrl IDC_loadIndicatorBar) progressSetPosition (loadVest GVAR(center));
+
+        vestContainer GVAR(center)
+    };
+    // Backpack
+    case IDC_buttonBackpack: {
+        if (_addOrRemove) then {
+            for "_i" from 1 to ([1, 5] select GVAR(shiftState)) do {
+                GVAR(center) addItemToBackpack _item;
+            };
+        } else {
+            for "_i" from 1 to ([1, 5] select GVAR(shiftState)) do {
+                GVAR(center) removeItemFromBackpack _item;
             };
+        };
 
-        _load = loadBackpack GVAR(center);
-        _maxLoad = backpack GVAR(center);
-        _items = backpackItems GVAR(center);
-        GVAR(currentItems) set [17,_items];
+        _containerItems = backpackItems GVAR(center);
+
+        // Update currentItems
+        GVAR(currentItems) set [IDX_CURR_BACKPACK_ITEMS, ((getUnitLoadout GVAR(center)) select IDX_LOADOUT_BACKPACK) param [1, []]];
+
+        // Update load bar
+        (_display displayCtrl IDC_loadIndicatorBar) progressSetPosition (loadBackpack GVAR(center));
+
+        backpackContainer GVAR(center)
     };
 };
 
-// Update progress bar status, weight info
-private _loadIndicatorBarCtrl = _display displayCtrl IDC_loadIndicatorBar;
-_loadIndicatorBarCtrl progressSetPosition _load;
-
-private _value = {_x == _item} count _items;
-_ctrlList lnbSetText [[_lnbCurSel, 2],str _value];
+// Find out how many items of that type there are and update the number displayed
+_ctrlList lnbSetText [[_lnbCurSel, 2], str ({_x == _item} count _containerItems)];
 
+//////////////////// Return NUMBER, as it did before or return BOOL, as the doc says is should ?
```

Comment:
As the comments asks: What should be done? As it is now it returns a bool, previously it returned a number. Doc says it returns bool.